log/format.go

```diff
@@ -95,348 +30,180 @@ type TerminalStringer interface {
 	TerminalString() string
 }
 
-// TerminalFormat formats log log.Records optimized for human readability on
-// a terminal with color-coded level output and terser human friendly timestamp.
-// This format should only be used for interactive programs or while developing.
-//
-//	[LEVEL] [TIME] MESSAGE key=value key=value ...
-//
-// Example:
-//
-//	[DBUG] [May 16 20:58:45] remove route ns=haproxy addr=127.0.0.1:50002
-func TerminalFormat(usecolor bool) Format {
-	return FormatFunc(func(r *log.Record) []byte {
-		msg := escapeMessage(r.Msg)
-		color := 0
-		if usecolor {
-			switch r.Lvl {
-			case log.LvlCrit:
-				color = 35
-			case log.LvlError:
-				color = 31
-			case log.LvlWarn:
-				color = 33
-			case log.LvlInfo:
-				color = 32
-			case log.LvlDebug:
-				color = 36
-			case log.LvlTrace:
-				color = 34
-			}
+func (h *TerminalHandler) format(buf []byte, r slog.Record, usecolor bool) []byte {
+	msg := escapeMessage(r.Message)
+	var color = ""
+	if usecolor {
+		switch r.Level {
+		case LevelCrit:
+			color = "\x1b[35m"
+		case slog.LevelError:
+			color = "\x1b[31m"
+		case slog.LevelWarn:
+			color = "\x1b[33m"
+		case slog.LevelInfo:
+			color = "\x1b[32m"
+		case slog.LevelDebug:
+			color = "\x1b[36m"
+		case LevelTrace:
+			color = "\x1b[34m"
 		}
+	}
+	if buf == nil {
+		buf = make([]byte, 0, 30+termMsgJust)
+	}
+	b := bytes.NewBuffer(buf)
+
+	// Note the timestamp is moved before the log level compared to upstream
+	b.WriteString("[")
+	writeTimeTermFormat(b, r.Time)
+	b.WriteString("] ")
```

Comment:
note this change as well (this would be the most difficult to use the upstream code for)